runtime/gc_base/GCExtensions.cpp

```diff
@@ -331,9 +330,14 @@ MM_GCExtensions::needScanStacksForContinuationObject(J9VMThread *vmThread, j9obj
 	 *
 	 * we don't scan the continuation object before started and after finished - java stack does not exist.
 	 */
-	if (started && !finished) {
+	uintptr_t continuationState = continuation->state;
+	if (VM_VMHelpers::isStartedFromContinuationState(continuationState) && !VM_VMHelpers::isFinishedFromContinuationState(continuationState)) {
 		Assert_MM_true(NULL != continuation);
-		needScan = !VM_VMHelpers::isContinuationMountedOrConcurrentlyScanned(continuation, isGlobalGC);
+		if (beingMounted) {
+			needScan = !VM_VMHelpers::isConcurrentlyScannedFromContinuationState(continuationState, isGlobalGC);
+		} else {
+			needScan = !VM_VMHelpers::isContinuationFullyMountedOrConcurrentlyScanned(continuationState, isGlobalGC);
```

Comment:
similarly for start/mounted check, we should think if this is harmful to check this so early, specifically for concurrent scan, since we will check it later, as part of tryWin, anyway